### PR TITLE
feat(TripPlanner.Form): use Ecto.Schema to create a form

### PR DESCRIPTION
**Pull request by thecristen**
#### Summary of changes

**Asana Ticket:** a part of [Trip planner form `Phoenix.LiveComponent`](https://app.asana.com/0/555089885850811/1207828528645185/f)

It's my first foray using Ecto, and really my first time in-depth with working with forms in Phoenix in general, but I've been working on a form with this schema as a base! This only includes the from and to locations for now, and the leave/arrive choice, mode selection, and wheelchair: true choices will be added later. Open to feedback on how I might improve this. 

- For each location (origin/destination) I wanted to be able to capture the coordinates, name, and stop ID (if a stop is chosen), since these are what get passed into OTP.
- I had a lot of trouble getting the validation to do what I wanted, so I ended up writing a custom one that adds an error to the location field if any of those sub-fields were invalid/missing. But I'd really rather have been able to make `cast_embed(:from, required: true, required_message: "Please select an origin")` or its `invalid_message` counterpart work.

---

#### General checks
* [x] **Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?** This is a good practice that can facilitate easier reviews.
* [x] **Testing.** Do the changes include relevant passing updates to tests? This includes updating screenshots. Preferably tests are run locally to verify that there are no test failures created by these changes, before opening a PR.
* [x] **Tech debt.** Have you checked for tech debt you can address in the area you're working in? This can be a good time to address small issues, or create Asana tickets for larger issues.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -143,6 +145,7 @@ defmodule DotCom.Mixfile do
       {:telemetry_poller, "1.1.0"},
       {:telemetry_test, "0.1.2", only: [:test]},
       {:timex, "3.7.11"},
+      {:typed_ecto_schema, "0.4.1"},
```
[the remark this answers is not in the document]
Comment (thecristen) [reply mix.exs:148]: I'm honestly not sure what the benefits are for this particular lib yet but I'm presuming it'll be more apparent down the line :P

[the remark this answers is not in the document]
Comment (anthonyshull) [reply mix.exs:148]: It generates the type so you don't have to have a schema and a type.

File: lib/trip_planner/input_form.ex [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    def changeset(struct \\ %__MODULE__{}, params \\ %{}) do
+      struct
+      |> cast(params, [:latitude, :longitude, :name, :stop_id])
+      |> add_default_name()
+      |> validate_required([:latitude, :longitude])
+    end
+
+    defp add_default_name(changeset) do
+      struct = apply_changes(changeset)
```
[the remark this answers is not in the document]
Comment (thecristen) [reply lib/trip_planner/input_form.ex:88]: so while trying this I'm realizing that that will only work if the fields in question happen to have changed, which won't always be the case 🤔

[the remark this answers is not in the document]
Comment (thecristen) [reply lib/trip_planner/input_form.ex:88]: okay I've got it, cool

**Review by bklebe — APPROVED** [review]

**Pull request merged**